### PR TITLE
Client csharp

### PR DESCRIPTION
**Pull request by giorgiozoppi**
Adding support for incubator-iotdb csharp .NET core client.

### CODE REVIEW

**Comment by qiaojialin** [comment]
Thanks! These files are failed in rat check. Please add apache header.

client-csharp/IoTDBClient/IoTDBClient.csproj
client-csharp/IoTDBService.csproj
client-csharp/README.md

**Comment by giorgiozoppi** [comment]
Fixed the headers.

File: client-csharp/IoTDBClient/Program.cs [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        TS_2DIFF,
+        BITMAP,
+        GORILLA,
+        REGULAR
+    }
+    public enum Compressor: ushort { UNCOMPRESSED, SNAPPY, GZIP, LZO, SDT, PAA, PLA}
+
+    class Program
+    {
+
```
Comment (xingtanzjr) [inline client-csharp/IoTDBClient/Program.cs:53]: Let's remove duplicated blank lines here : )

File: client-csharp/IoTDBClient/Program.cs [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        static async Task Main(string[] args)
+        {
+            var port = 6667;
+            var username = "root";
+            var password = "root";
+            
+            IoTDBService.TSIService.Client client = null;
+            try
+            {
+
```
Comment (xingtanzjr) [inline client-csharp/IoTDBClient/Program.cs:64]: Let's remove the blank line here : )

**Comment by jixuan1989** [comment]
Hi, as there is no guy approved the pr, I will learn some knowledge of Csharp and then try this PR. 
need some time.

**Comment by giorgiozoppi** [comment]
Ok. Let me do a refactor before

**Comment by jixuan1989** [comment]
Hi @giorgiozoppi ,

you can add the following codes in root/pom.xml (search "<artifactId>maven-thrift-plugin</artifactId>" and you will know where to put the codes to,), 

```
<execution>
                                <id>generate-thrift-sources-csharp</id>
                                <phase>generate-sources</phase>
                                <goals>
                                    <goal>compile</goal>
                                </goals>
                                <configuration>
                                    <generator>csharp</generator>
                                    <thriftExecutable>${thrift.exec.absolute.path}</thriftExecutable>
                                    <thriftSourceRoot>${basedir}/src/main/thrift</thriftSourceRoot>
                                    <outputDirectory>${project.build.directory}/generated-sources-csharp</outputDirectory>
                                </configuration>
                            </execution>
```

Then, running  `mvn generate-sources -DskipTests -pl service-rpc` (if failed, just use `mvn package -DskipTests`), you will get *.cs files under root/service-rpc/target/generated-sources-csharp. 

It will be better than using`compile.sh` because in this way, users do not need to install thrift (maven will download the thrift compiler if needed).

**Comment by giorgiozoppi** [comment]
We move this inside the CI like @chrisdutz  suggested.

**Comment by giorgiozoppi** [comment]
we stash close this.

**Pull request closed without merging**